File: Cano-Rodriguez-Claudio/Procesos/Go/queue_reading.go
package main

import (
	"fmt"
	"os"
	"syscall"
	"unsafe"
)

type Message struct {
	Type int64
	Text [100]byte
}

func main() {
	key, _ := syscall.Ftok("progfile", 65)

	// Create message queue
	msgid, _, err := syscall.Syscall(syscall.SYS_MSGGET, uintptr(key), 0666|syscall.IPC_CREAT, 0)
	if err != 0 {
		fmt.Println("Error creating message queue")
		os.Exit(1)
	}

	var message Message

	// Receive message
	_, _, err = syscall.Syscall6(syscall.SYS_MSGRCV, msgid, uintptr(unsafe.Pointer(&message)), 100, 1, 0, 0)
	if err != 0 {
		fmt.Println("Error receiving message")
		os.Exit(1)
	}

	fmt.Printf("Data received is: %s\n", string(message.Text[:]))

	// Remove message queue
	syscall.Syscall(syscall.SYS_MSGCTL, msgid, syscall.IPC_RMID, 0)
}

File: Cano-Rodriguez-Claudio/Procesos/Java/queue_reading.java
import java.io.*;
import java.util.Scanner;

public class SignalR {
    public static void main(String[] args) {
        File file = new File("message.txt");

        try {
            // Simulate receiving the message
            Scanner reader = new Scanner(file);
            String message = reader.nextLine();
            System.out.println("Data received is: " + message);
            reader.close();

            // Simulate removing the queue (delete the file)
            if (file.delete()) {
                System.out.println("Message queue removed successfully.");
            } else {
                System.err.println("Error removing the message queue.");
            }
        } catch (IOException e) {
            System.err.println("Error reading message: " + e.getMessage());
        }
    }
}
